plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystemMinio.java

```diff
@@ -0,0 +1,175 @@
+        try (TrinoS3FileSystem fs = new TrinoS3FileSystem()) {
+            fs.initialize(new URI(testBucketPath), s3Configuration());
+
+            AmazonS3 s3 = fs.getS3Client();
+
+            assertThat(listPaths(s3, testBucketName, "", true)).isEmpty();
+
+            fs.delete(new Path(testBucketPath), false);
+
+            assertThat(listPaths(s3, testBucketName, "", true)).isEmpty();
+        }
+    }
+
+    @Test
+    public void testDeleteNonRecursivelyNonEmptyBucketRoot()
+            throws Exception
+    {
+        String testBucketName = "trino-delete-bucket-root-non-empty" + randomBucketSuffix();
+        minioClient.makeBucket(testBucketName);
+        String testBucketPath = "s3://%s/".formatted(testBucketName);
+        try (TrinoS3FileSystem fs = new TrinoS3FileSystem()) {
+            fs.initialize(new URI(testBucketPath), s3Configuration());
+
+            AmazonS3 s3 = fs.getS3Client();
+            fs.createNewFile(new Path("s3://%s/file1.txt".formatted(testBucketName)));
+            String directory2Path = testBucketPath + "directory2";
+            createDirectory(fs.getS3Client(), testBucketName, "directory2");
+            String filename2 = "file2.txt";
+            fs.createNewFile(new Path(directory2Path, filename2));
+
+            assertThat(listPaths(s3, testBucketName, "", true))
+                    .containsOnly("file1.txt", "directory2/", "directory2/file2.txt");
+
+            assertThatThrownBy(() -> fs.delete(new Path(testBucketPath), false))
+                    .hasMessage("Directory %s is not empty".formatted(testBucketPath));
+
+            assertThat(listPaths(s3, testBucketName, "", true))
+                    .containsOnly("file1.txt", "directory2/", "directory2/file2.txt");
+        }
+    }
+
+    @Test
+    public void testDeleteRecursivelyBucketRoot()
+            throws Exception
+    {
+        String testBucketName = "trino-delete-recursive-bucket-root" + randomBucketSuffix();
+        minioClient.makeBucket(testBucketName);
+        String testBucketPath = "s3://" + testBucketName;
+        try (TrinoS3FileSystem fs = new TrinoS3FileSystem()) {
+            fs.initialize(new URI(testBucketPath), s3Configuration());
+
+            AmazonS3 s3 = fs.getS3Client();
+            fs.createNewFile(new Path("s3://%s/file1.txt".formatted(testBucketName)));
+            String directory2Path = testBucketPath + "/directory2";
+            createDirectory(fs.getS3Client(), testBucketName, "directory2");
+            fs.createNewFile(new Path(directory2Path, "file2.txt"));
+
+            assertThat(listPaths(s3, testBucketName, "", true))
+                    .containsOnly("file1.txt", "directory2/", "directory2/file2.txt");
+
+            assertTrue(fs.delete(new Path(testBucketPath + Path.SEPARATOR), true));
+
+            assertThat(listPaths(s3, testBucketName, "", true)).isEmpty();
+        }
+    }
+
+    private static String randomBucketSuffix()
```

Comment:
In other places we shamelessly abuse `randomTableSuffix`

https://github.com/trinodb/trino/blob/389403af94c28e3c6c17ebdca0c95bae2bd170e4/plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java#L79

instead of adding (many) clones of this method, let's rename it -- https://github.com/trinodb/trino/pull/15008